### PR TITLE
Refactor resourcecreator (WIP)

### PR DESCRIPTION
**Pull request by VegarM**
The idea here is to make the code related to creating different resources more manageable by:
1. Splitting the code in `resourcecreator.go` into different files (grouped by what kind of resource they manage) 
2. Create higher order structures **asap** in order to prevent passing redundant information to every function just to create a higher order structure way down the call hierarchy. (i.e `ObjectMeta`)

Split out progress:
- [x] service
- [x] redis
- [ ] deployment
- [ ] secret
- [ ] ingress
- [ ] autoscaler 
- [ ] alerts

### CODE REVIEW

File: api/resourcecreator.go [hunk 1]
```diff
@@ -516,7 +490,9 @@ func createAutoscalerSpec(min, max, cpuTargetPercentage int, application string)
 func createOrUpdateK8sResources(deploymentRequest NaisDeploymentRequest, manifest NaisManifest, resources []NaisResource, clusterSubdomain string, istioEnabled bool, k8sClient kubernetes.Interface) (DeploymentResult, error) {
 	var deploymentResult DeploymentResult
 
-	service, err := createService(deploymentRequest, manifest.Team, k8sClient)
+	objectMeta := createObjectMeta(deploymentRequest.Application, deploymentRequest.Namespace, manifest.Team)
```
Comment (VegarM) [inline api/resourcecreator.go]: Here and only here should we create the `ObjectMeta` struct

**Comment by gtcno** [comment]
+1 for splitting. 

Consider moving into a appropriate package? 
A generic "api"  folder/package doesnt say much.

**Comment by VegarM** [comment]
Agreed, should be in its own package.

File: api/objectmetacreator_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	t.Run("Test required metadata field values", func(t *testing.T) {
+		objectMeta := createObjectMeta(appName, namespace, teamName)
+		objectMetaWithoutTeamName := createObjectMeta(appName, namespace, "")
+
+		assert.Equal(t, teamName, objectMeta.Labels["team"], "Team label should be equal to team name.")
+		assert.Equal(t, appName, objectMeta.Labels["app"], "App label should be equal to app name.")
+		assert.Equal(t, appName, objectMeta.Name, "Resource name should equal app name.")
+		assert.Equal(t, namespace, objectMeta.Namespace, "Resource namespace should equal namespace.")
+
+		_, ok := objectMetaWithoutTeamName.Labels["team"]
```
Comment (audunstrand) [inline api/objectmetacreator_test.go]: dette burde være en egen test

**Comment by audunstrand** [comment]
I sugget mergin to master bewteen each, if possible

**Comment by VegarM** [comment]
Yup, will delete this branch and start over.
A few of the decisions made in this branch are sub-optimal.

**Pull request closed without merging**